dbms/src/Formats/PrettySpaceBlockOutputStream.cpp

```diff
@@ -26,12 +25,6 @@ void PrettySpaceBlockOutputStream::write(const Block & block)
     Widths name_widths;
     calculateWidths(block, widths, max_widths, name_widths, format_settings);
 
-    /// Do not align on too long values.
```

Comment:
This block affects only appearance, but it breaks the invariant "max_width[i] >= widths[i][j] for each j for each i" that leads to infinite loop at https://github.com/yandex/ClickHouse/blob/master/dbms/src/Formats/PrettyBlockOutputStream.cpp#L202

Subtracting unsigned integers (size_t) is problematic because it can easily lead to overflow (but here it makes this bug visible).

Reply:
But I want to save this behaviour.

Reply:
It is really important.

Reply:
Could you explain in more details the behaviour?

Reply:
Or I can just change result cycling loop from "a < b - c" to "a + c < b" here https://github.com/yandex/ClickHouse/pull/5560#discussion_r291810162 . It would save the behavior without infinite loop. Is it good in your opinion?

Reply:
As we have this logic:

https://github.com/yandex/ClickHouse/blob/master/dbms/src/Formats/PrettyBlockOutputStream.cpp#L64

the code you deleted is really obsolete.